system/src/control/config.cpp

```diff
@@ -55,8 +60,203 @@ namespace control {
 
 namespace config {
 
+namespace {
+
 using namespace particle::control::common;
 
+struct SecurityModeChangeContext {
+    char clientNonce[32];
+};
+
+std::unique_ptr<SecurityModeChangeContext> g_securityModeChangeCtx;
+bool g_securityModeChangePending = false;
+
+inline void setSecurityModeOverride() {
+    HAL_Core_Write_Backup_Register(BKP_DR_08, 1); // Disable security
+}
+
+inline void clearSecurityModeOverride() {
+    HAL_Core_Write_Backup_Register(BKP_DR_08, 0); // Use default security mode
+}
+
+inline bool isSecurityModeOverridden() {
+    return HAL_Core_Read_Backup_Register(BKP_DR_08);
+}
+
+// Completion handler for system_ctrl_set_result()
+void systemResetCompletionHandler(int result, void* data) {
+    system_reset(SYSTEM_RESET_MODE_NORMAL, RESET_REASON_CONFIG_UPDATE, 0 /* value */, 0 /* flags */, nullptr /* reserved */);
```

Comment:
Will this reset the device after the unlock request succeeds? Can this be avoided? We want to support use cases like gathering logs from a running protected device without resetting it.